test/Pass/single-pass.ll

```diff
@@ -6,7 +6,7 @@
 define i32 @foo(i32 %x) {
 entry:
   %add = add nsw i32 %x, 1
-  ; CHECK-NOT: icmp
+  ; CHECK: icmp
```

Comment:
with this change you're looking for an icmp but an icmp already existed in the test case, so the test becomes useless. the point of this test is to look for an optimization happening. it's no longer shown by eliminating the icmp, you need to look at something different.

Reply:
I have fixed it. It tests the current functionality that instruction is not deleted but the uses are replaced by value, so we don't look for the uses of instruction now.

Reply:
I think it'd be better to check for the whole line we want:
%conv = zext i1 true to i32
(or whatever it really is)